climada/hazard/tc_tracks.py

```diff
@@ -1029,22 +1029,11 @@ def to_geodataframe(self, as_points=False):
         )
 
         if as_points:
-            times = np.concatenate([track.time.data for track in self.data])
-            points = np.concatenate([
-                np.c_[track.lon, track.lat] for track in self.data
-            ])
-
-            # repeat each idx according to the number of timesteps
-            lens = [track.time.size for track in self.data]
-            idx = np.repeat(gdf.index.to_numpy(), lens)
-
-            gdf_long = gpd.GeoDataFrame({
-                'idx': idx,
-                'time': times,
-                'geometry': [Point(p[0], p[1]) for p in points],
-            }).set_index('idx')
-
-            gdf = gdf.join(gdf_long)
+            gdf_long = pd.concat([track.to_dataframe().assign(idx=i) for i, track in enumerate(self.data)])
+            gdf_long['geometry'] = gdf_long.apply(lambda x: Point(x['lon'],x['lat']), axis=1)
+            gdf_long = gdf_long.drop(columns=['lon', 'lat'])
+            gdf_long = gpd.GeoDataFrame(gdf_long.reset_index().set_index('idx'), geometry='geometry')
+            gdf = gdf_long.join(gdf)
```

Comment:
Why does the join work here? Do the dataframes made from the tracks not have much less rows than the one made point the track points?

Reply:
> Do the dataframes made from the tracks not have much less rows than the one made point the track points?
Yes, exactly. So for the point case, you want to provide the track attributes for each point, hence the join.

Reply:
Ok, sounds correct. I probably misunderstood how join is working.